### PR TITLE
Uncommon remote scheduled task created

### PR DESCRIPTION
**Pull request by OmriItzhak**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-11998?filter=72118

## Description
New playbook and trigger to handle the alert "Uncommon remote scheduled task created".
Enhancement to command line analysis playbook - added new check for suspicious path and suspicious arguments from the command line.
 
## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Core/ReleaseNotes/3_0_86.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+For response actions, the following integrations are required: 
+
+- PAN-OS.
+
+
+#### Triggers Recommendations
+
+##### New: Uncommon remote scheduled task created
+
+- New: This trigger is responsible for handling 'Uncommon remote scheduled task created.
```
Comment (julieschwartz18) [inline Packs/Core/ReleaseNotes/3_0_86.md:36]: This trigger runs the **Uncommon remote scheduled task created** Playbook to handle these alerts.

File: Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml [hunk 2]
```diff
@@ -0,0 +1,1352 @@
+id: Uncommon remote scheduled task created
+version: -1
+contentitemexportablefields:
+  contentitemfields: {}
+name: Uncommon remote scheduled task created
+description: "This playbook is designed to handle the alert \"Uncommon remote scheduled task created\".\n\nPlaybook Stages:\n\nAnalysis:\n\n- The playbook checks if the remote IP is external or has a bad reputation.\n\nInvestigation:\nDuring the alert investigation, the playbook will perform the following:\n\n- Searches for related XDR alerts to the endpoint using the following MITRE techniques to identify any malicious activity: T1202 - Indirect Command Execution, T1021 - Remote Services.\n- Searches for related XSIAM agent alerts on the remote endpoint to determine if the creation of the scheduled task is part of an attack pattern.\n- Searches for suspicious command-line parameters indicating a malicious scheduled task.\n\nRemediation:\n\n- Remediation actions will disable the scheduled task and block the malicious IP, and close the alert.\n\nRequirements:\n\nFor response actions, the following integrations are required: \n\n- PAN-OS."
```
Comment (julieschwartz18) [inline Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml:6]: Recommend:
 "This playbook handles \"Uncommon remote scheduled task created\" alerts.\n\nPlaybook Stages:\n\nAnalysis:\n\n- The playbook checks if the remote IP is external or has a bad reputation.\n\nInvestigation:\nThe playbook searches for the following:\n\n- Related Cortex XDR alerts on the endpoint that use the following MITRE techniques to identify any malicious activity: T1202 - Indirect Command Execution, T1021 - Remote Services.\n- Related Cortex XSIAM agent alerts on the remote endpoint, to determine if the creation of the scheduled task is part of an attack pattern.\n- Suspicious command line parameters indicating a malicious scheduled task.\n\nRemediation:\n\n- Disable the scheduled task,  block the malicious IP, and close the alert.\n\nRequirements:\n\nFor response actions, the following integrations are required: \n\n- PAN-OS."

File: Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml [hunk 3]
```diff
@@ -0,0 +1,1352 @@
...
+    type: condition
+    task:
+      id: eae7099d-0e36-4442-8d50-a5e79d067791
+      version: -1
+      name: Check whether the remote IP is malicious or external
+      description: |-
+        Determines the appropriate verdict based on:
+        - IP Reputation.
+        - Whether the task was created from an external IP address.
+        - Suspicious Command-line parameters.
```
Comment (julieschwartz18) [inline Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml:85]: Suspicious command line parameters.

File: Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml [hunk 4]
```diff
@@ -0,0 +1,1352 @@
...
+    isautoswitchedtoquietmode: false
+  "22":
+    id: "22"
+    taskid: 25929bfd-f6cd-43f9-87cd-8d0c0caf677d
+    type: regular
+    task:
+      id: 25929bfd-f6cd-43f9-87cd-8d0c0caf677d
+      version: -1
+      name: Disable the malicious scheduled task manually
+      description: "Dear Analyst,\n\nPlease note that during the remediation process, the playbook failed to disable the scheduled task: ${Core.OriginalAlert.event.scheduled_task_path}\n\nPlease take manual action to disable the scheduled task. "
```
Comment (julieschwartz18) [inline Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml:502]: "Dear Analyst,\n\nDuring the remediation process the playbook failed to disable the following scheduled task: ${Core.OriginalAlert.event.scheduled_task_path}\n\nPlease manually disable this scheduled task. "

File: Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml [hunk 5]
```diff
@@ -0,0 +1,1352 @@
...
+    isautoswitchedtoquietmode: false
+  "42":
+    id: "42"
+    taskid: 951c96f5-d9b9-4a75-802a-da1d01d66810
+    type: condition
+    task:
+      id: 951c96f5-d9b9-4a75-802a-da1d01d66810
+      version: -1
+      name: Analyst decision for suspicious cmd parameters
+      description: Analyst review is required to determine, based on suspicious command-line parameters, whether to take remediation actions  such as disabling the scheduled task and blocking the IP if malicious or to close the alert as a false positive.
```
Comment (julieschwartz18) [inline Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml:925]: Based on suspicious command line parameters, analyst review is required to determine whether to take remediation actions such as disabling the scheduled task and blocking the IP if malicious, or to close the alert as a false positive.

File: Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml [hunk 6]
```diff
@@ -0,0 +1,1352 @@
...
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    message:
+      to:
+      subject:
+      body:
+        simple: |-
+          The following command-line contains a suspicious argument:
```
Comment (julieschwartz18) [inline Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml:951]: The following command line contains a suspicious argument:

File: Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml [hunk 7]
```diff
@@ -0,0 +1,1352 @@
...
+  "47":
+    id: "47"
+    taskid: 4373ba97-486c-4617-8298-86a924dc5ca8
+    type: regular
+    task:
+      id: 4373ba97-486c-4617-8298-86a924dc5ca8
+      version: -1
+      name: Search for related alerts by name and MITRE Technique
+      description: |+
+        This task searches for suspicious alerts related to alert by MITRE techniques that may indicate a compromised endpoint.
```
Comment (julieschwartz18) [inline Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml:1108]: This task searches by MITRE technique for suspicious alerts related to the alert that may indicate a compromised endpoint.

File: Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml [hunk 8]
```diff
@@ -0,0 +1,1352 @@
...
+      scriptName: Print
+      type: regular
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "23"
+    scriptarguments:
+      value:
+        simple: "Dear Analyst,\n\nPlease note that during the remediation process, the playbook executed a shell command to disable the following scheduled task: \n${Core.OriginalAlert.event.scheduled_task_path}\n\n"
```
Comment (julieschwartz18) [inline Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml:1270]: "Dear Analyst,\n\nDuring the remediation process the playbook executed a shell command to disable the following scheduled task: \n${Core.OriginalAlert.event.scheduled_task_path}\n\n"

**Review by julieschwartz18 — CHANGES_REQUESTED** [review]
TW review done

**Comment by julieschwartz18** [comment]
Finished TW review with recommended edits. In the meantime, checked docs-approved.

File: Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created_README.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+### Commands
+
+* closeInvestigation
+* core-get-cloud-original-alerts
+* core-get-script-execution-results
+* core-run-script-execute-commands
+* ip
+
+## Playbook Inputs
+
```
Comment (altmannyarden) [inline Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created_README.md:56]: lets remove all the empty sections in the readme

**Review by altmannyarden — APPROVED** [review]

File: Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml [hunk 10]
```diff
@@ -0,0 +1,1433 @@
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "32":
+    id: "32"
+    taskid: 789cf6e0-eded-4b32-8108-8091409a2537
+    type: condition
+    task:
+      id: 789cf6e0-eded-4b32-8108-8091409a2537
+      version: -1
+      name: Found any alerts indicating this is a malicious scheduled task?
```
Comment (altmannyarden) [inline Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml:772]: I think this should be renamed. as you search fir malware category alerts, not specific to a schedule task

File: Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml [hunk 11]
```diff
@@ -0,0 +1,1433 @@
...
+    message:
+      to:
+      subject:
+      body:
+        simple: |-
+          The following command-line contains a suspicious argument:
+
+          ${Core.OriginalAlert.event.scheduled_task_image_command_line}
+
+          Do you want to proceed and disable the scheduled task?
```
Comment (altmannyarden) [inline Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml:955]: missing a lot of context. 
like my comment above, please elaborate on the alert situation

File: Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml [hunk 12]
```diff
@@ -0,0 +1,1433 @@
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "47":
+    id: "47"
+    taskid: 4373ba97-486c-4617-8298-86a924dc5ca8
+    type: regular
+    task:
+      id: 4373ba97-486c-4617-8298-86a924dc5ca8
+      version: -1
+      name: Search for related alerts by name and MITRE Technique
```
Comment (altmannyarden) [inline Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml:1106]: on which host? the remote one? the other one?

File: Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml [hunk 13]
```diff
@@ -0,0 +1,1433 @@
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "2":
+    id: "2"
+    taskid: eae7099d-0e36-4442-8d50-a5e79d067791
+    type: condition
+    task:
+      id: eae7099d-0e36-4442-8d50-a5e79d067791
+      version: -1
+      name: Check whether the remote IP is malicious or external
```
Comment (altmannyarden) [inline Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml:80]: is malicious or external? will internal IPs return as malicious in a reputation check?
Please rename this check

Comment (altmannyarden) [reply Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml:80]: To be honest, I dont understand the condition. lets talk about it

File: Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created_README.md [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+Investigation:
+During the alert investigation, the playbook will perform the following:
+
+- Searches for related XSIAM alerts on the endpoint that use the following MITRE techniques to identify malicious activity: T1202 - Indirect Command Execution, T1021 - Remote Services.
+- Searches for related XSIAM agent alerts on the remote endpoint, to determine if the creation of the scheduled task is part of an attack pattern.
+- Searches for suspicious command-line parameters indicating a malicious scheduled task.
+
+Remediation:
+
+- Remediation actions will disable the scheduled task, block the malicious IP, and close the alert.
```
[the remark this answers is not in the document]
Comment (altmannyarden) [reply Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created_README.md:18]: worth adding what will require analyst approval and what will run automatically

**Review by julieschwartz18 — APPROVED** [review]

File: Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml (hunk 10 above)
Comment (OmriItzhak) [reply Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml:772]: changed to: "Found any alerts of malicious activity on the remote host?"

File: Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created_README.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (OmriItzhak) [reply Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created_README.md:18]: Done

File: Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml (hunk 13 above)
Comment (OmriItzhak) [reply Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml:80]: Done

File: Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml (hunk 12 above)
Comment (OmriItzhak) [reply Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml:1106]: Done

File: Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml (hunk 13 above)
Comment (OmriItzhak) [reply Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml:80]: Done

File: Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml [hunk 15]
```diff
@@ -0,0 +1,1433 @@
...
+    isautoswitchedtoquietmode: false
+  "42":
+    id: "42"
+    taskid: 951c96f5-d9b9-4a75-802a-da1d01d66810
+    type: condition
+    task:
+      id: 951c96f5-d9b9-4a75-802a-da1d01d66810
+      version: -1
+      name: Analyst decision for suspicious cmd parameters
+      description: Based on suspicious command line parameters, analyst review is required to determine whether to take remediation actions such as disabling the scheduled task and blocking the IP if malicious, or to close the alert as a false positive.
```
[the remark this answers is not in the document]
Comment (OmriItzhak) [reply Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml:925]: Done

File: Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created_README.md (hunk 9 above)
Comment (OmriItzhak) [reply Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created_README.md]: Done

File: Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml (hunk 15 above)
[the remark this answers is not in the document]
Comment (OmriItzhak) [reply Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml]: Done

File: Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml (hunk 11 above)
Comment (OmriItzhak) [reply Packs/Core/Playbooks/playbook-Uncommon_remote_scheduled_task_created.yml]: Done

**Review by umishkin — APPROVED** [review]

**Pull request merged**